### PR TITLE
Fix method for finding the installed share/opae directory.

### PR DESCRIPTION
**Pull request by michael-adler**
The true installation path is unknown when users install from an RPM.
All we can count on is the relative positions of the installed
bin and share directories.

Also fixed an error in finding all the platforms and interfaces for the
--help message.

These are both needed for release/0.13.

### CODE REVIEW

**Comment by michael-adler** [comment]
Can someone please approve this so it makes its way to release/0.13 today?

File: platforms/scripts/afu_platform_config [hunk 1]
```diff
@@ -782,17 +784,17 @@ def findPlatforms(db_path):
...
     # Walk all the directories
     for db_dir in db_path:
-        try:
-            # Look for JSON files in each directory
-            for fn in os.listdir(db_dir):
-                if fn.endswith('.json'):
+        # Look for JSON files in each directory
+        for fn in os.listdir(db_dir):
+            if fn.endswith('.json'):
+                try:
```
Comment (r-rojo) [inline platforms/scripts/afu_platform_config]: Would it be better to use glob?
Something like:
```python
for db_dir in  db_path:
    for json_file in glob.glob(os.path.join(db_dir, "*.json")):
...
```

File: platforms/scripts/afu_platform_config [hunk 2]
```diff
@@ -804,19 +806,19 @@ def findAfuIfcs(db_path):
...
```
Comment (r-rojo) [inline platforms/scripts/afu_platform_config]: Same thing here, use glob?
Also, would it make sense to refactor the process of finding json files in a directory tree?

File: platforms/scripts/afu_platform_config [hunk 3]
```diff
@@ -804,19 +806,19 @@ def findAfuIfcs(db_path):
...
                         # If it has a module-arguments entry assume the file is
                         # valid
                         if ('module-arguments' in db):
                             afus.add(fn[:-5])
-        except Exception:
-            # Give up on this file or directory if there is any error
-            None
+                except Exception:
+                    # Give up on this file or directory if there is any error
+                    None
```
Comment (r-rojo) [inline platforms/scripts/afu_platform_config]: Do we want to give up on a directory if there is an error in just one JSON file? Wouldn't it be better to give up on just that file? Perhaps a little more explicit exception handling could help?

File: platforms/scripts/afu_platform_config [hunk 4]
```diff
@@ -782,17 +784,17 @@ def findPlatforms(db_path):
...
                     with open(os.path.join(db_dir, fn)) as f:
                         # Does it have a platform name field?
                         db = json.load(f)
                         platforms.add(db['platform-name'])
-        except Exception:
-            # Give up on this file or directory if there is any error
-            None
+                except Exception:
+                    # Give up on this file or directory if there is any error
+                    None
```
Comment (r-rojo) [inline platforms/scripts/afu_platform_config]: Do we want to give up on a directory if there is an error in just one JSON file? Wouldn't it be better to give up on just that file? Perhaps a little more explicit exception handling could help?

**Comment by michael-adler** [comment]
@r-rojo - all of these are fine points but unrelated to the commit and the immediate problem that release/0.13 is freezing and the commit is fixing a bug already present when an rpm's --prefix is moved. Unless you have a problem with the specific change, please approve. I'll review the other comments later today or tomorrow.

**Comment by r-rojo** [comment]
Sounds good.

File: platforms/scripts/afu_platform_config (hunk 3 above)
Comment (michael-adler) [reply platforms/scripts/afu_platform_config]: It was the old code that gave up on the directory when one error was found. That's fixed in the commit, though I failed to change the comment. I will.

**Pull request merged**